exporters/webhook/app.py

```diff
@@ -214,11 +219,19 @@ async def pelorus_webhook(
     )
 
 
-@app.get("/metrics", response_class=PlainTextResponse)
+@app.get("/{path:path}", response_class=PlainTextResponse)
 async def metrics():
     return generate_latest()
 
 
+@app.route("/{path:path}", methods=["*"])
```

Comment:
I was not able to trigger this, is this necessary?

Commenting it and running `curl -X POST http://localhost:8080/wrong -d @./webhook_pelorus_committime.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: committime"` also throws a 405

Reply:
Of course, because you are trying POST to the /wrong, replace /wrong with /pelorus/webhook and you will receive metric.

Reply:
no no, I am trying to trigger the `HTTPException` from `catch_all_not_allowed`, doing wrong stuff

The GET falls to the metrics, so it has to be a different HTTP verb

Reply:
```
$ curl -X POST http://localhost:8080/wrong \
        -H "User-Agent: Pelorus-Webhook/test" \
        -H "X-Pelorus-Event: committime" \
        -H "Content-Type: application/json" \
         -d @./webhook_pelorus_committime.json 

{"detail":"Method Not Allowed"}
```



```
$ curl -X GET http://localhost:8080/wrong \
        -H "User-Agent: Pelorus-Webhook/test" \
        -H "X-Pelorus-Event: committime" \
        -H "Content-Type: application/json" \
         -d @./webhook_pelorus_committime.json 
         
       
# HELP python_gc_objects_collected_total Objects collected during gc
# TYPE python_gc_objects_collected_total counter
python_gc_objects_collected_total{generation="0"} 258.0
python_gc_objects_collected_total{generation="1"} 355.0
python_gc_objects_collected_total{generation="2"} 0.0
# HELP python_gc_objects_uncollectable_total Uncollectable object found during GC
# TYPE python_gc_objects_uncollectable_total counter
python_gc_objects_uncollectable_total{generation="0"} 0.0
python_gc_objects_uncollectable_total{generation="1"} 0.0
python_gc_objects_uncollectable_total{generation="2"} 0.0
# HELP python_gc_collections_total Number of times this generation was collected
# TYPE python_gc_collections_total counter
python_gc_collections_total{generation="0"} 201.0
python_gc_collections_total{generation="1"} 18.0
python_gc_collections_total{generation="2"} 1.0
# HELP python_info Python platform information
# TYPE python_info gauge
python_info{implementation="CPython",major="3",minor="10",patchlevel="9",version="3.10.9"} 1.0
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 2.98827776e+08
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 6.621184e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.67865236657e+09
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 2.58
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 9.0
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1024.0
# HELP webhook_received_total Number of received webhooks
# TYPE webhook_received_total counter
webhook_received_total 0.0
# HELP webhook_received_created Number of received webhooks
# TYPE webhook_received_created gauge
webhook_received_created 1.6786523691208856e+09
# HELP webhook_processed_total Number of processed webhooks
# TYPE webhook_processed_total counter
webhook_processed_total 0.0
# HELP webhook_processed_created Number of processed webhooks
# TYPE webhook_processed_created gauge
webhook_processed_created 1.6786523691209576e+09
# HELP commit_timestamp Commit timestamp
# TYPE commit_timestamp gauge
# HELP deploy_timestamp Deployment timestamp
# TYPE deploy_timestamp gauge
# HELP failure_creation_timestamp Failure Creation Timestamp
# TYPE failure_creation_timestamp gauge
# HELP failure_resolution_timestamp Failure Resolution Timestamp
# TYPE failure_resolution_timestamp gauge
```

```
$ curl -X GET http://localhost:8080/wrong \
        -H "User-Agent: Pelorus-Webhook/test" \
        -H "X-Pelorus-Event: committime" \
        -H "Content-Type: application/json" \
         -d @./webhook_pelorus_committime.json 

{"http_response":"Webhook Received","http_response_code":200
```

```
$ curl -X POST http://localhost:8080/wrong -d @./webhook_pelorus_committime.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: committime"

{"detail":"Method Not Allowed"}
```

Everything as expected, last one is yours...

On the server side, it's ok:
```
INFO:     Started server process [342250]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:     127.0.0.1:49744 - "POST /wrong HTTP/1.1" 405 Method Not Allowed
INFO:     127.0.0.1:57426 - "GET /wrong HTTP/1.1" 200 OK
INFO:     127.0.0.1:52804 - "POST /pelorus/webhook HTTP/1.1" 202 Accepted
INFO:     127.0.0.1:56912 - "POST /wrong HTTP/1.1" 405 Method Not Allowed
INFO:     127.0.0.1:56928 - "POST /wrong HTTP/1.1" 405 Method Not Allowed
INFO:     127.0.0.1:33106 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:33106 - "GET /favicon.ico HTTP/1.1" 200 OK
INFO:     127.0.0.1:33114 - "GET /metrics HTTP/1.1" 200 OK
INFO:     127.0.0.1:33114 - "GET /favicon.ico HTTP/1.1" 200 OK
```

Reply:
So, that's my question

I have run these commands both with your code and commenting lines 227 to 232, and I get the same results.

Are lines 227 to 232 of the code necessary?

Reply:
You are correct, looks like it's FastAPI default so no need to duplicate.

Reply:
Actually this is coming from starlette, but same idea.